### DISCUSSION TITLE
Vampire drain:  keep track of overall loss, ignoring any gain from charging cycles

### DISCUSSION THREAD

**Discussion by wishbone1138** [teslamate-org/teslamate#429] [Ideas]
Confused by vampire drain reporting. It feels like it should be:

From drive state D-P to next drive state P-D where not charging or pre-heating?

However, the reporting behavior seems almost random for me. 

Here is an example of my recent states:
![image](https://user-images.githubusercontent.com/2423249/74056750-cca1b980-499f-11ea-8229-e5165400987b.png)

However, the only one reported over 6 hours period is the time on 2/5:
![image](https://user-images.githubusercontent.com/2423249/74056785-dfb48980-499f-11ea-99be-56bb6db44b24.png)

I might say that the system is ignoring sleep time but then the 2/5 entry doesn't appear to make much sense.

**Comment by wishbone1138** [comment]
I think I may have answered my own question.  It appears that if there is a charging state at all during that time, then it will ignore that entire period?  Is there any way to keep track of overall loss, ignoring any gain from charging cycles?

**Comment by adriankumpf** [comment]
> It appears that if there is a charging state at all during that time, then it will ignore that entire period?

Yes, this is exactly how the current implementation or database query works.

> Is there any way to keep track of overall loss, ignoring any gain from charging cycles?

We have the data for that. The query "just" needs to be modified accordingly.

**Comment by wishbone1138** [comment]
Got it, I know it's not trivial, I've tried to calculate that myself in the past but my database fu is lacking.  Would this be a better way of doing it?  I can take a look if you, and others, agree.

**Comment by ArnoPW** [comment]
Besides ignoring gain from charging cycles it could also be an option to stop monitoring once the heater goes on. Now if you preheat the car there's a rather high vampire drain due to the preheating of the car.

**Comment by rasssta** [comment]
Somewhat related to to this issue; I'd like an additional graph showing overall vampire drain vs. driving usage similar to the AC vs DC charging comparison (x % vampire drain, y % driving). Not sure if this should be a new issue or integrated into this one?
